### PR TITLE
Backport pipeline improvements for EL7/8/MicroOS

### PR DESCRIPTION
**Pull request by andypitcher**
Improve pipeline and scripts for EL7, EL8 and MicroOS.

- Remove dapper
- Add sign --dry-run
- Add repo-metadata script

### CODE REVIEW

File: policy/centos7/scripts/build [hunk 1]
```diff
@@ -4,6 +4,9 @@ set -e -x
 cd $(dirname $0)/..
 . ./scripts/version
 
+yum install -y epel-release
+yum install -y container-selinux selinux-policy-devel yum-utils rpm-build # git
```
Comment (macedogm) [inline policy/centos7/scripts/build:8]:
```suggestion
yum install -y container-selinux selinux-policy-devel yum-utils rpm-build
```

File: policy/centos7/scripts/upload-repo [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
-yum install -y git python2-pip python-deltarpm
-pip install boto3==1.17.112
-pip install --cache-dir=/var/cache/pip/ \
-  git+git://github.com/Voronenko/rpm-s3.git@5695c6ad9a08548141d3713328e1bd3f533d137e
+# Package required to unzip awscliv2.zip
+yum install -y unzip
 
+# Install the awscli-v2 from AWS
+curl "https://awscli.amazonaws.com/awscli-exe-linux-$(uname -m).zip" -o "awscliv2.zip"
+unzip awscliv2.zip
```
Comment (macedogm) [inline policy/centos7/scripts/upload-repo:12]:
```suggestion
unzip -q awscliv2.zip
```

Comment (macedogm) [reply policy/centos7/scripts/upload-repo:12]: To make the unzip output quite and remove noise from the CI logs.

**Review by macedogm — CHANGES_REQUESTED** [review]
Can you please also remove the following Docker volume lines that were only needed when using Dapper's Docker-in-Docker build?

**Part 1**
https://github.com/rancher/rancher-selinux/blob/e449419dc13e6fc66d6db8dd1f36a744d77bda15/.drone.yml#L14-L16

**Part 2**
https://github.com/rancher/rancher-selinux/blob/e449419dc13e6fc66d6db8dd1f36a744d77bda15/.drone.yml#L95-L98

**Review by macedogm — CHANGES_REQUESTED** [review]
Some further changes (that I can't directly point in the PR):

1. In:
https://github.com/rancher/rancher-selinux/blob/686bc8593b812f06b53d1c4cec2cefb6dbe8c7c1/.drone.yml#L167

Need to match the same change done for EL7 -> `- "policy/centos8/dist/**/*.rpm"`

2. In:
https://github.com/rancher/rancher-selinux/blob/686bc8593b812f06b53d1c4cec2cefb6dbe8c7c1/.drone.yml#L353

Need to match the same change done for EL7 -> `- "policy/microos/dist/**/*.rpm"`

**Comment by andypitcher** [comment]
@macedogm last [commit](https://github.com/rancher/rancher-selinux/pull/27/commits/9b76fbf5d020a4bf5d9fe756fb0adb259835abba) implements your last [recommendation](https://github.com/rancher/rancher-selinux/pull/27#pullrequestreview-1706086357)..

**Review by macedogm — APPROVED** [review]

**Pull request merged**